src/lava/lib/optimization/utils/generators/mis.py

```diff
@@ -0,0 +1,180 @@
+    def connection_prob(self) -> float:
+        """Returns the connection probability of the graph."""
+        return self._connection_prob
+
+    @property
+    def seed(self):
+        """Returns the seed used for generating the graph."""
+        return self._seed
+
+    @staticmethod
+    def _get_graph_from_adjacency_matrix(adjacency_matrix):
+        num_vertices = adjacency_matrix.shape[0]
+        # create Graph
+        G = netx.Graph()
+        G.add_nodes_from(range(num_vertices))
+
+        for v1 in range(num_vertices):
+            for v2 in range(v1 + 1, num_vertices):
+                if adjacency_matrix[v1, v2] == 1:
+                    G.add_edge(v1, v2)
+        return G
+
+    @staticmethod
+    def _get_adjacency_of_complement_graph(
+            adjacency_matrix: np.ndarray) -> np.ndarray:
+        adj_mc = np.logical_not(adjacency_matrix)
+        adj_mc = adj_mc.astype(int)
+        adj_mc = adj_mc - np.diag(adj_mc.diagonal())
+        return adj_mc
+
+    @staticmethod
+    def _get_qubo_cost_from_adjacency(adjacency_matrix: np.ndarray,
+                                      w_diag: float,
+                                      w_off: float) -> np.ndarray:
+        if w_off <= 2 * w_diag:
+            raise ValueError(
+                "Off-diagonal weights must be > 2 x diagonal weights.")
+
+        # Translate the connectivity matrix to a QUBO matrix
+        num_variables = adjacency_matrix.shape[0]
+        q = - w_diag * np.eye(num_variables) + w_off / 2 * adjacency_matrix
+        return q.astype(int)
+
+    def get_graph(self) -> netx.Graph:
+        """Returns the graph in networkx format."""
+        graph = self._get_graph_from_adjacency_matrix(self._adjacency)
+        return graph
+
+    def get_graph_matrix(self) -> np.ndarray:
+        """Returns the adjacency matrix of the graph."""
+        return self._adjacency
+
+    def get_complement_graph(self) -> netx.Graph:
+        """Returns the complement graph in networkx format."""
+        c_adjacency = self._get_adjacency_of_complement_graph(self._adjacency)
+        c_graph = self._get_graph_from_adjacency_matrix(c_adjacency)
+        return c_graph
+
+    def get_complement_graph_matrix(self) -> np.ndarray:
+        """Returns the adjacency matrix of the complement graph."""
+        c_adjacency = self._get_adjacency_of_complement_graph(self._adjacency)
+        return c_adjacency
+
+    def get_as_qubo(self, w_diag: float, w_off: float) -> QUBO:
+        """
+        Creates a QUBO whose solution corresponds to the maximum independent
+        set (MIS) of the graph defined by the input adjacency matrix.
+
+        The goal of the QUBO is to minimize the cost
+            min x^T * Q * x ,
+        where the vector x is defined as:
+            x_i = 1 if vertex i is part of the MIs
+            x_i = 0 if vertex i is not part of the MIS,
+        and the QUBO matrix is given by
+            Q_ii = w_diag
+            Q_ij = w_off (for i~=j) .

```

Comment:
In Python, not equal is "!=". You probably spent too much time with Matlab...? :-D